### PR TITLE
fix: drop sync after receiving push notifications

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

App doesn't receive notifications after a while in the Background.

### Causes

Kalium has some issues that are fixed in the PRs mentioned above.
It also introduces `ConnectionPolicy` so we can make it stop Sync if the app is in the background.

### Solutions

`ConnectionPolicyManager` to observe the app lifecycle.
If the UI was never initialised, then set all sessions to `DISCONNECT_AFTER_PENDING_EVENTS`.

If the UI was initialised, then set the current active session to `KEEP_ALIVE`.

Just observing Background/Foreground (with onPause and onResume) can lead to sync starting and stopping unnecessarily as the user just opens another app for a couple of seconds.

### Dependencies

Needs releases with:

- https://github.com/wireapp/kalium/pull/707
- https://github.com/wireapp/kalium/pull/706

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Open the app, verify that you have Sync going on.
Force-close the application, send yourself a message.
App should receive a notifications and keep receiving in the future (delay between notifications caused by SlowSync in the background).

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-794/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3955980)(3955980)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3955980)(3955980)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt [hunk 1]
```diff
@@ -148,6 +148,11 @@ class UseCaseModule {
     fun provideObserveSyncStateUseCase(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: UserId) =
         coreLogic.getSessionScope(currentAccount).observeSyncState
 
+    @ViewModelScoped
+    @Provides
+    fun provideSetConnectionPolicyUseCase(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: UserId) =
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:153]: is this provider used somewhere?

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 2]
```diff
@@ -9,6 +9,7 @@ import com.wire.kalium.logic.CoreLogic
 import com.wire.kalium.logic.data.id.ConversationId
 import com.wire.kalium.logic.data.id.QualifiedID
 import com.wire.kalium.logic.data.notification.LocalNotificationConversation
+import com.wire.kalium.logic.data.sync.ConnectionPolicy
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt:12]: is code-style checking okay about not used imports???

File: app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt [hunk 3]
```diff
@@ -40,9 +42,7 @@ class HomeViewModel @Inject constructor(
         private set
 
     init {
-        viewModelScope.launch {
-            launch { loadUserAvatar() }
-        }
+        viewModelScope.launch { loadUserAvatar() }
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt:45]: basically all the changes in this file could be reverted

**Review by borichellow — APPROVED** [review]
nice one! 🔥  just a few small comments

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt (hunk 2 above)
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt:12]: Not in `Reloaded`, only in `Kalium`.

I'd say we can enable it here too... @yamilmedina what do you think? I can go ahead with this.

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt (hunk 1 above)
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:153]: Aaah yeah, I forgot to take this one out.

I had a different implementation earlier that I ended up not using. I'll take it out.

File: app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt (hunk 3 above)
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt:45]: Indeed.

Just this `viewmodelScope.launch` tiny optimisation, that avoids running a unnecessary `Job` within a `Job`.

I'll revert these.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-794/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3981181)(3981181)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3981181)(3981181)-dev-release.apk)

**Pull request merged**